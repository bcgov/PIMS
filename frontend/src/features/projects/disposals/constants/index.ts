export * from './defaultFilter';
export * from './defaultProjectForm';
