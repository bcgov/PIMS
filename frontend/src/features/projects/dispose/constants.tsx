import { DisposeWorkflowStatus, ProjectWorkflowComponent } from '.';
import React from 'react';

//TODO: map to real components
export const projectWorkflowComponents: ProjectWorkflowComponent[] = [
  { component: () => <p>Step 1</p>, workflowStatus: DisposeWorkflowStatus.Draft },
  { component: () => <p>Step 2</p>, workflowStatus: DisposeWorkflowStatus.SelectProperties },
  { component: () => <p>Step 3</p>, workflowStatus: DisposeWorkflowStatus.UpdateInformation },
  { component: () => <p>Step 4</p>, workflowStatus: DisposeWorkflowStatus.RequiredDocumentation },
  { component: () => <p>Step 5</p>, workflowStatus: DisposeWorkflowStatus.Approval },
  { component: () => <p>Step 6</p>, workflowStatus: DisposeWorkflowStatus.Submitted },
];
