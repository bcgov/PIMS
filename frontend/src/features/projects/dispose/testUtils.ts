import {
  IProjectTask,
  DisposeWorkflowStatus,
  IProject,
  ReviewWorkflowStatus,
  ITask,
  SPPApprovalTabs,
  NoteTypes,
} from '../common';
import { ProjectActions } from 'constants/actionTypes';
import * as reducerTypes from 'constants/reducerTypes';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import * as API from 'constants/API';

export const mockTasks: IProjectTask[] = [
  {
    projectNumber: 123,
    taskId: 1,
    isOptional: true,
    isCompleted: false,
    name: 'task-0',
    description: 'one',
    taskType: 1,
    sortOrder: 0,
    completedOn: new Date(),
    statusId: 0,
    statusCode: DisposeWorkflowStatus.RequiredDocumentation,
  },
  {
    projectNumber: 123,
    taskId: 2,
    isOptional: true,
    isCompleted: true,
    name: 'task-1',
    description: 'two',
    taskType: 1,
    sortOrder: 0,
    completedOn: new Date(),
    statusId: 0,
    statusCode: DisposeWorkflowStatus.RequiredDocumentation,
  },
];

export const mockProject: IProject = {
  projectNumber: 'test-01',
  name: 'my project',
  description: 'my project description',
  privateNote: 'private note',
  publicNote: 'public note',
  notes: [
    {
      noteType: NoteTypes.General,
      note: 'general',
    },
    {
      noteType: NoteTypes.Public,
      note: 'public',
    },
    {
      noteType: NoteTypes.Private,
      note: 'private',
    },
    {
      noteType: NoteTypes.Exemption,
      note: 'exemption',
    },
    {
      noteType: NoteTypes.AgencyInterest,
      note: 'agencyinterest',
    },
    {
      noteType: NoteTypes.Financial,
      note: 'financial',
    },
    {
      noteType: NoteTypes.PreMarketing,
      note: 'premarketing',
    },
    {
      noteType: NoteTypes.Marketing,
      note: 'marketing',
    },
    {
      noteType: NoteTypes.ContractInPlace,
      note: 'contractinplace',
    },
    {
      noteType: NoteTypes.Reporting,
      note: 'reporting',
    },
    {
      noteType: NoteTypes.LoanTerms,
      note: 'loanterms',
    },
    {
      noteType: NoteTypes.Adjustment,
      note: 'adjustment',
    },
    {
      noteType: NoteTypes.SppCost,
      note: 'sppcost',
    },
    {
      noteType: NoteTypes.SppGain,
      note: 'sppgain',
    },
    {
      noteType: NoteTypes.SalesHistory,
      note: 'saleshistory',
    },
    {
      noteType: NoteTypes.CloseOut,
      note: 'closeout',
    },
    {
      noteType: NoteTypes.Comments,
      note: 'comments',
    },
  ],
  properties: [],
  agencyId: 1,
  statusId: 0,
  statusCode: ReviewWorkflowStatus.ApprovedForErp,
  status: {
    id: 1,
    name: 'Approved for ERP',
    sortOrder: 0,
    route: '',
    description: '',
    workflowCode: '',
    code: ReviewWorkflowStatus.ApprovedForErp,
    isMilestone: true,
    tasks: [],
    isOptional: false,
    isActive: true,
  },
  tierLevelId: 1,
  tasks: mockTasks,
  note: 'my notes',
  id: 1,
  fiscalYear: 2020,
  projectAgencyResponses: [],
};

export const tasks: ITask[] = [
  {
    taskId: 1,
    name: 'task-0',
    sortOrder: 0,
    description: 'test',
    taskType: 1,
  },
  {
    taskId: 2,
    name: 'task-1',
    sortOrder: 0,
    description: 'test',
    taskType: 1,
  },
];

export const mockWorkflow = [
  {
    description:
      'A new draft project that is not ready to submit to apply to be added to the Surplus Property Program.',
    route: '/projects/draft',
    isMilestone: false,
    code: 'DR',
    id: 1,
    name: 'Draft',
    isDisabled: false,
    sortOrder: 0,
    isOptional: false,
    workflowCode: 'SUBMIT-DISPOSAL',
    type: 'ProjectStatus',
    createdOn: '2020-06-29T04:46:45.2666667',
    rowVersion: 'AAAAAAAACUM=',
    isActive: true,
    tasks: [],
  },
  {
    description: 'Add properties to the project.',
    route: '/projects/properties',
    isMilestone: false,
    code: 'DR-P',
    id: 2,
    name: 'Select Properties',
    isDisabled: false,
    sortOrder: 1,
    isOptional: false,
    workflowCode: 'SUBMIT-DISPOSAL',
    type: 'ProjectStatus',
    createdOn: '2020-06-29T04:46:45.2666667',
    rowVersion: 'AAAAAAAACUQ=',
    isActive: true,
    tasks: [],
  },
  {
    description: 'Assign tier level, classification and update current financial information.',
    route: '/projects/information',
    isMilestone: false,
    code: 'DR-I',
    id: 3,
    name: 'Update Information',
    isDisabled: false,
    sortOrder: 2,
    isOptional: false,
    workflowCode: 'SUBMIT-DISPOSAL',
    type: 'ProjectStatus',
    createdOn: '2020-06-29T04:46:45.2666667',
    rowVersion: 'AAAAAAAACUU=',
    isActive: true,
    tasks: [],
  },
  {
    description:
      'Required documentation has been completed and sent (Surplus Declaration \u0026 Readiness Checklist, Triple Bottom Line).',
    route: '/projects/documentation',
    isMilestone: false,
    code: 'DR-D',
    id: 4,
    name: 'Required Documentation',
    isDisabled: false,
    sortOrder: 3,
    isOptional: false,
    workflowCode: 'SUBMIT-DISPOSAL',
    type: 'ProjectStatus',
    createdOn: '2020-06-29T04:46:45.2666667',
    rowVersion: 'AAAAAAAACUY=',
    isActive: true,
    tasks: [],
  },
  {
    description: 'The project is ready to be approved by owning agency.',
    route: '/projects/approval',
    isMilestone: false,
    code: 'DR-A',
    id: 5,
    name: 'Approval',
    isDisabled: false,
    sortOrder: 4,
    isOptional: false,
    workflowCode: 'SUBMIT-DISPOSAL',
    type: 'ProjectStatus',
    createdOn: '2020-06-29T04:46:45.2666667',
    rowVersion: 'AAAAAAAACUc=',
    isActive: true,
    tasks: [],
  },
  {
    description:
      'The project has been submitted for review to be added to the Surplus Property Program.',
    route: '/projects/review',
    isMilestone: false,
    code: 'DR-RE',
    id: 6,
    name: 'Review',
    isDisabled: false,
    sortOrder: 5,
    isOptional: false,
    workflowCode: 'SUBMIT-DISPOSAL',
    type: 'ProjectStatus',
    createdOn: '2020-06-29T04:46:45.2666667',
    rowVersion: 'AAAAAAAACUg=',
    isActive: true,
    tasks: [],
  },
];

const mockStore = configureMockStore([thunk]);
export const getStore = (mockProject: IProject, tab?: SPPApprovalTabs) =>
  mockStore({
    [reducerTypes.LOOKUP_CODE]: {
      lookupCodes: [
        {
          code: 'BCT',
          name: 'BC Transit',
          id: 1,
          type: API.AGENCY_CODE_SET_NAME,
        },
      ],
    },
    [reducerTypes.ProjectReducers.PROJECT]: { project: mockProject },
    [reducerTypes.ProjectReducers.TASKS]: tasks,
    [reducerTypes.NETWORK]: {
      [ProjectActions.GET_PROJECT]: {},
    },
    [reducerTypes.ProjectReducers.SPL_TAB]: tab,
  });
