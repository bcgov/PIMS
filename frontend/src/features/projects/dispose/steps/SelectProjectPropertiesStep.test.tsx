import { cleanup, fireEvent, render, waitFor } from '@testing-library/react';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { ProjectActions } from 'constants/actionTypes';
import * as API from 'constants/API';
import * as reducerTypes from 'constants/reducerTypes';
import { createMemoryHistory } from 'history';
import { mockFlatProperty } from 'mocks/filterDataMock';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import { act } from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { mockKeycloak, mockProject } from '../testUtils';
import SelectProjectProperties from './SelectProjectPropertiesStep';
jest.mock('@react-keycloak/web');

const mockStore = configureMockStore([thunk]);
const history = createMemoryHistory();
const mockAxios = new MockAdapter(axios);

const store = mockStore({
  [reducerTypes.ProjectReducers.PROJECT]: { project: mockProject },
  [reducerTypes.LOOKUP_CODE]: {
    lookupCodes: [{ name: 'agencyVal', id: 1, isDisabled: false, type: API.AGENCY_CODE_SET_NAME }],
  },
  [reducerTypes.NETWORK]: {
    requests: { [ProjectActions.GET_PROJECT]: {} },
  },
});

const uiElement = (
  <Provider store={store}>
    <Router history={history}>
      <SelectProjectProperties />
    </Router>
  </Provider>
);

describe('Select Project Properties Step', () => {
  beforeEach(() => {
    mockAxios.reset();
    mockKeycloak([], [1]);
  });
  afterEach(() => {
    cleanup();
  });
  it('renders correctly', async () => {
    mockAxios.onAny().reply(200, { items: [mockFlatProperty] });
    await act(async () => {
      const { container, findByText } = render(uiElement);
      expect(container.firstChild).toMatchSnapshot();
      await findByText('Test Property');
    });
  });

  it('requests properties based on the users agency', async () => {
    mockAxios.onAny().reply(200, { items: [mockFlatProperty] });
    render(uiElement);
    await waitFor(async () => {
      expect(mockAxios.history.get.length).toBe(1);
      expect(mockAxios.history.get[0].url).toMatch(/agencies=1/);
    });
  });
  it('initially requests properties using a default classification id', async () => {
    mockAxios.onAny().reply(200, { items: [mockFlatProperty] });
    render(uiElement);
    await waitFor(async () => {
      expect(mockAxios.history.get.length).toBe(1);
      expect(mockAxios.history.get[0].url).toMatch(/classificationId=2/);
    });
  });
  it('requests properties that may already be in another project', async () => {
    mockAxios.onAny().reply(200, { items: [mockFlatProperty] });
    render(uiElement);
    await waitFor(async () => {
      expect(mockAxios.history.get.length).toBe(1);
      expect(mockAxios.history.get[0].url).toMatch(/ignorePropertiesInProjects=false/);
    });
  });

  it('renders one row per property in the response', async () => {
    mockAxios.onAny().reply(200, { items: [mockFlatProperty] });
    const { findByText } = render(uiElement);
    await waitFor(async () => {
      const propertyText = await findByText('Test Property');
      expect(propertyText).toBeInTheDocument();
    });
  });

  it('concatenates the address and admin area', async () => {
    mockAxios.onAny().reply(200, { items: [mockFlatProperty] });
    const { findByText } = render(uiElement);
    await waitFor(async () => {
      const propertyText = await findByText('1234 Test St, Victoria');
      expect(propertyText).toBeInTheDocument();
    });
  });
  it('displays all associated projects', async () => {
    const propertyWithProjects = {
      ...mockFlatProperty,
      projectNumbers: ['SPP-10000', 'SPP-10001'],
    };
    mockAxios.onAny().reply(200, { items: [propertyWithProjects] });
    const { findByText } = render(uiElement);
    await waitFor(async () => {
      const projectOneText = await findByText('SPP-10000');
      const projectTwoText = await findByText('SPP-10001');
      expect(projectOneText).toBeInTheDocument();
      expect(projectOneText).toHaveAttribute('href', '/projects?projectNumber=SPP-10000');
      expect(projectTwoText).toBeInTheDocument();
      expect(projectTwoText).toHaveAttribute('href', '/projects?projectNumber=SPP-10001');
    });
  });
  it('allows properties to be added to the project', async () => {
    mockAxios.onAny().reply(200, { items: [mockFlatProperty] });
    const { findByTestId, findByText, findAllByText, getByText } = render(uiElement);
    await waitFor(async () => {
      const selectRowCheck = await findByTestId('selectrow-0');
      fireEvent.click(selectRowCheck);
    });
    const addToProjectBtn = await findByText('Add To Project');
    fireEvent.click(addToProjectBtn);
    const propertyNameText = await findAllByText('Test Property');
    const selectedText = getByText('1 Selected');
    expect(selectedText).toBeInTheDocument();
    expect(propertyNameText).toHaveLength(2);
  });
  it('requests a new page of data when a paging button is pressed', async () => {
    mockAxios.onAny().reply(200, { items: [mockFlatProperty], total: 6 });
    const { findByText, getByLabelText } = render(uiElement);
    await findByText('Test Property');
    const pageButton = getByLabelText('Page 2');
    fireEvent.click(pageButton);
    await waitFor(async () => {
      expect(mockAxios.history.get.length).toBe(2);
      expect(mockAxios.history.get[1].url).toMatch(/page=2/);
    });
  });
  it('selected properties are maintained even if the page changes.', async () => {
    mockAxios.onAny().reply(200, { items: [mockFlatProperty], total: 6 });
    const { findByTestId, findByText, findAllByText, getByLabelText, getByText } = render(
      uiElement,
    );

    //select a property
    const selectRowCheck = await findByTestId('selectrow-0');
    fireEvent.click(selectRowCheck);
    //change the page (with the property still selected)
    const pageButton = getByLabelText('Page 2');
    fireEvent.click(pageButton);
    //add all selected properties to the project.
    const addToProjectBtn = await findByText('Add To Project');
    fireEvent.click(addToProjectBtn);
    //ensure that the property is in both tables.
    const propertyNameText = await findAllByText('Test Property');
    const selectedText = getByText('1 Selected');
    expect(selectedText).toBeInTheDocument();
    expect(propertyNameText).toHaveLength(2);
  });

  it('selected properties are still checked even if the page changes.', async () => {
    mockAxios.onAny().reply(200, { items: [{ ...mockFlatProperty, id: 1 }], total: 6 });
    const { findByTestId, getByLabelText } = render(uiElement);

    //select a property
    let selectRowCheck = await findByTestId('selectrow-1');
    fireEvent.click(selectRowCheck);
    //change the page (with the property still selected)
    let pageButton = getByLabelText('Page 2');
    fireEvent.click(pageButton);
    //change back to page 1.
    pageButton = getByLabelText('Page 1');
    fireEvent.click(pageButton);
    //ensure that the property is in both tables.
    selectRowCheck = await findByTestId('selectrow-1');
    await waitFor(async () => {
      expect(selectRowCheck).toHaveAttribute('checked', '');
    });
  });

  it('selected properties can be removed', async () => {
    mockAxios.onAny().reply(200, { items: [mockFlatProperty] });
    const { findByTestId, findByText, findAllByText, getByTitle } = render(uiElement);
    await waitFor(async () => {
      const selectRowCheck = await findByTestId('selectrow-0');
      fireEvent.click(selectRowCheck);
    });
    const addToProjectBtn = await findByText('Add To Project');
    fireEvent.click(addToProjectBtn);
    const removeButton = getByTitle('Click to remove from project');
    fireEvent.click(removeButton);

    const propertyNameText = await findAllByText('Test Property');
    expect(propertyNameText).toHaveLength(1);
  });
});
