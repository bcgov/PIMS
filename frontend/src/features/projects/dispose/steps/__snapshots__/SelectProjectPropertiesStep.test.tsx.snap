// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="SelectProjectProperties container-fluid"
>
  <h3
    className="mr-auto"
  >
    Search and select 1 or more properties for the project
  </h3>
  <div
    className="filter-container border-bottom container-fluid"
  >
    <div
      className="px-0 container"
    >
      <form
        className=""
        noValidate={true}
        onSubmit={[Function]}
      >
        <div
          className="filter-bar form-row"
        >
          <div
            className="bar-item col"
          >
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="input-pid"
                name="pid"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                placeholder="Enter PID or PIN"
                value=""
              />
            </div>
          </div>
          <div
            className="bar-item col"
          >
            <div
              className="input-group"
            >
              <div
                className="input-group-content"
              >
                <div
                  className="form-group"
                >
                  <input
                    className="form-control"
                    id="input-address"
                    name="address"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                    placeholder="Enter an address or city"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
          <div
            className="bar-item col"
          >
            <div
              className="form-group"
            >
              <select
                className="form-select form-control"
                id="input-agencies"
                name="agencies"
                onBlur={[Function]}
                onChange={[Function]}
                value=""
              >
                <option
                  value=""
                >
                  Enter an Agency
                </option>
              </select>
            </div>
          </div>
          <div
            className="bar-item col"
          >
            <div
              className="form-group"
            >
              <select
                className="form-select form-control"
                id="input-classificationId"
                name="classificationId"
                onBlur={[Function]}
                onChange={[Function]}
                value=""
              >
                <option
                  value=""
                >
                  Classification
                </option>
              </select>
            </div>
          </div>
          <div
            className="bar-item flex-grow-0 col"
          >
            <button
              className="Button Button--icon-only bg-warning btn btn-primary"
              disabled={false}
              onBlur={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="submit"
            >
              <div
                className="Button__icon"
              >
                <svg
                  fill="currentColor"
                  height={20}
                  size={20}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 512 512"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                  />
                </svg>
              </div>
            </button>
          </div>
          <div
            className="bar-item flex-grow-0 col"
          >
            <button
              className="Button Button--icon-only btn btn-info"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              type="reset"
            >
              <div
                className="Button__icon"
              >
                <svg
                  fill="currentColor"
                  height={20}
                  size={20}
                  stroke="currentColor"
                  strokeWidth="0"
                  style={
                    Object {
                      "color": undefined,
                    }
                  }
                  viewBox="0 0 512 512"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M212.333 224.333H12c-6.627 0-12-5.373-12-12V12C0 5.373 5.373 0 12 0h48c6.627 0 12 5.373 12 12v78.112C117.773 39.279 184.26 7.47 258.175 8.007c136.906.994 246.448 111.623 246.157 248.532C504.041 393.258 393.12 504 256.333 504c-64.089 0-122.496-24.313-166.51-64.215-5.099-4.622-5.334-12.554-.467-17.42l33.967-33.967c4.474-4.474 11.662-4.717 16.401-.525C170.76 415.336 211.58 432 256.333 432c97.268 0 176-78.716 176-176 0-97.267-78.716-176-176-176-58.496 0-110.28 28.476-142.274 72.333h98.274c6.627 0 12 5.373 12 12v48c0 6.627-5.373 12-12 12z"
                  />
                </svg>
              </div>
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
  <form
    className=""
    noValidate={true}
    onSubmit={[Function]}
  >
    <div
      className="col-md-12 PropertyListViewSelect container"
    >
      <div
        className="ScrollContainer"
      >
        <div
          className="TableToolbar container-fluid"
        >
          <h2
            className="mr-auto"
          >
            Available Properties
          </h2>
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Add Selected
          </button>
        </div>
        <div
          className="table"
          role="table"
          style={
            Object {
              "minWidth": undefined,
            }
          }
        >
          <div
            className="thead thead-light"
          >
            <div
              className="tr"
              role="row"
              style={
                Object {
                  "display": "flex",
                  "flex": "1 0 auto",
                  "minWidth": "675px",
                }
              }
            >
              <div
                className="th reset-filter svg-btn"
              />
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "40 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "40px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  <div>
                    <input
                      checked={false}
                      onChange={[Function]}
                      style={
                        Object {
                          "cursor": "pointer",
                        }
                      }
                      title="Toggle All Rows Selected"
                      type="checkbox"
                    />
                  </div>
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "100 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "100px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Agency
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "100 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "100px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Name
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "100 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "100px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Civic Address
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "140 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "140px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Classification
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "100 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "100px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Current Zoning Code
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "100 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "100px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Potential Zoning Code
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "145 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "145px",
                    "textAlign": "left",
                    "width": "145px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Net Book Value
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "145 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "145px",
                    "textAlign": "left",
                    "width": "145px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Market Value
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "145 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "145px",
                    "textAlign": "left",
                    "width": "145px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Assessed Value
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "60 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "60px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Type
                </div>
              </div>
            </div>
          </div>
          <div
            className="no-rows-message"
          >
            No rows to display
          </div>
        </div>
        <div
          className="table-toolbar"
        />
      </div>
      <div
        className="ScrollContainer"
      >
        <div
          className="TableToolbar container-fluid"
        >
          <h2
            className="mr-auto"
          >
            Properties in the Project
          </h2>
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Remove Selected
          </button>
        </div>
        <div
          className="table"
          role="table"
          style={
            Object {
              "minWidth": undefined,
            }
          }
        >
          <div
            className="thead thead-light"
          >
            <div
              className="tr"
              role="row"
              style={
                Object {
                  "display": "flex",
                  "flex": "1 0 auto",
                  "minWidth": "705px",
                }
              }
            >
              <div
                className="th reset-filter svg-btn"
              />
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "40 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "40px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  <div>
                    <input
                      checked={false}
                      onChange={[Function]}
                      style={
                        Object {
                          "cursor": "pointer",
                        }
                      }
                      title="Toggle All Rows Selected"
                      type="checkbox"
                    />
                  </div>
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "40 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "40px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "100 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "100px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Agency
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "100 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "100px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Name
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "100 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "100px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Civic Address
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "140 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "140px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Classification
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "100 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "100px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Current Zoning Code
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "100 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "100px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Potential Zoning Code
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "145 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "145px",
                    "textAlign": "left",
                    "width": "145px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Net Book Value
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "145 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "145px",
                    "textAlign": "left",
                    "width": "145px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Market Value
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "145 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "145px",
                    "textAlign": "left",
                    "width": "145px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Assessed Value
                </div>
              </div>
              <div
                className="th"
                colSpan={1}
                role="columnheader"
                style={
                  Object {
                    "alignItems": "center",
                    "boxSizing": "border-box",
                    "display": "flex",
                    "flex": "60 0 auto",
                    "justifyContent": "flex-start",
                    "minWidth": "30px",
                    "textAlign": "left",
                    "width": "60px",
                  }
                }
              >
                <div
                  className="sortable-column"
                >
                  Type
                </div>
              </div>
            </div>
          </div>
          <div
            className="no-rows-message"
          >
            No rows to display
          </div>
        </div>
        <div
          className="table-toolbar"
        />
      </div>
    </div>
    <div
      className="ProjectNotes form-row"
    >
      <div
        className="col-md-10 form-group col"
      >
        <label
          className="form-label"
          htmlFor="input-note"
        >
          Notes
        </label>
        <textarea
          className="col-md-auto form-control"
          id="input-note"
          name="note"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
        />
      </div>
    </div>
    <div
      style={
        Object {
          "textAlign": "right",
        }
      }
    />
  </form>
</div>
`;
